src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -2144,7 +2349,21 @@ mgp_error mgp_graph_detach_delete_vertex(struct mgp_graph *graph, mgp_vertex *ve
     if (!MgpGraphIsMutable(*graph)) {
       throw ImmutableObjectException{"Cannot remove a vertex from an immutable graph!"};
     }
-    const auto result = graph->impl->DetachRemoveVertex(&vertex->impl);
+    const auto result =
+        std::visit(memgraph::utils::Overloaded{
+                       [vertex](memgraph::query::DbAccessor *impl) {
+                         if (std::holds_alternative<memgraph::query::SubgraphVertexAccessor>(vertex->impl)) {
+                           throw std::logic_error{"Can't procede with operation due to inner type error."};
```

Comment:
"Inner type error" isn't something that lights my bulb. Explain why logic error happens (forbidden to detach delete vertices in subgraph view becase partial graph is visible).

Reply:
change this to assert